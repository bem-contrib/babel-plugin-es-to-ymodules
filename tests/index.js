const pluginTester = require('babel-plugin-tester')
const path = require('path')
const plugin = require('../lib')

pluginTester({
    plugin: plugin.default,
    pluginName : 'es-to-ymodules',
    fixtures: path.join(__dirname, 'fixtures'),
    tests : [
        {
            title : 'Thorw error, if module name not defined',
            code : 'export default {}',
            snapshot : false,
            error : /Module name required/
        },
        {
            title : 'No thorw module name error, if plain ymodule',
            code : '/** */\n modules.define',
            snapshot : true
        },
        {
            title : 'Patch modules with short arrow',
            code : `modules.define('a', (p, a) => p(a))`,
            snapshot : true
        }
    ]
});
