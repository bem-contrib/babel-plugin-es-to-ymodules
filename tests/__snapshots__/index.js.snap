// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`es-to-ymodules No thorw module name error, if plain ymodule: No thorw module name error, if plain ymodule 1`] = `
"
/** */
 modules.define

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

/** */
modules.define;
"
`;

exports[`es-to-ymodules Patch modules with short arrow: Patch modules with short arrow 1`] = `
"
modules.define('a', (p, a) => p(a))

      ↓ ↓ ↓ ↓ ↓ ↓

'use strict';

modules.define('a', (p, a) => {
  a = _extractDefaultValue(a);

  function _extractDefaultValue(obj) {
    return obj && obj.__esModule ? obj.default : obj;
  }

  p(a);
});
"
`;

exports[`es-to-ymodules Simple import: Simple import 1`] = `
"
import block from \\"m\\"

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _m = require(\\"m\\");

var _m2 = _interopRequireDefault(_m);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
"
`;

exports[`es-to-ymodules Skip BinaryExpression exports: Skip BinaryExpression exports 1`] = `
"
exports === 10

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

exports === 10;
"
`;

exports[`es-to-ymodules Skip UnaryExpression exports: Skip UnaryExpression exports 1`] = `
"
typeof exports

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

typeof exports;
"
`;
